### PR TITLE
feat(postcss): support postcss v8 with explict postcss installation

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

As we can't [fully migrated to postcss 8](https://github.com/nuxt/nuxt.js/pull/8408) until [postcss-preset-env supports v8](https://github.com/csstools/postcss-preset-env/issues/191)

This pr is for supporting postcss config in nuxt if user installs postcss packages explicitly, required packages:

**NOTE: postcss-preset-env is disabled in v8 config as it doesn't support postcss v8 for now.**

```sh
yarn add --dev css-loader@^5.0.0 postcss@^8.1.10 postcss-import@^13.0.0 postcss-loader@^4.1.0 postcss-url@^10.1.1
# or
npm i --save-dev css-loader@^5.0.0 postcss@^8.1.10 postcss-import@^13.0.0 postcss-loader@^4.1.0 postcss-url@^10.1.1
```

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/webpack/src/utils/postcss-v8.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
+import fs from 'fs'
```
Comment (clarkdo) [inline packages/webpack/src/utils/postcss-v8.js:1]: Most code is duplicate of `postcss.js`, this is more handy when we fully migrate to v8: just remove `postcss.js` and rename `postcss-v8.js`

File: packages/webpack/src/utils/style-loader.js [hunk 2]
```diff
@@ -12,8 +13,15 @@ export default class StyleLoader {
     this.perfLoader = perfLoader
     this.resolveModule = resolveModule
 
-    if (buildContext.options.build.postcss) {
-      this.postcssConfig = new PostcssConfig(buildContext)
+    const { postcss: postcssOptions } = buildContext.options.build
+    if (postcssOptions) {
+      const postcss = require(resolveModule('postcss'))
```
Comment (clarkdo) [inline packages/webpack/src/utils/style-loader.js:18]: Is this OK to use `require(resolveModule(...))` ?

Comment (pi0) [reply packages/webpack/src/utils/style-loader.js:19]: Seems good 👍

**Review by pi0 — APPROVED** [review]

**Comment by AndrewBogdanovTSS** [comment]
I try to use suggested list of dependencies, specifically:
`yarn add --dev css-loader@^5.0.0 postcss@^8.1.10 postcss-import@^13.0.0 postcss-loader@^4.1.0 postcss-url@^10.1.1`
and than install postcss-nested plugin v.^5.0.5 I get compilation error:
```
 ERROR  Failed to compile with 1 errors


 ERROR  in ./.nuxt/components/nuxt-error.vue?vue&type=style&index=0&lang=css&

Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
ValidationError: Invalid options object. PostCSS Loader has been initialized using an options object that does not match the API schema.
 - options.postcssOptions should be one of these:
   object { config?, … } | function
   -> Options to pass through to `Postcss`.
   Details:
    * options.postcssOptions.config should be one of these:
      string | boolean
      -> Allows to specify PostCSS Config Path (https://github.com/postcss/postcss-loader#config)
      Details:
       * options.postcssOptions.config should be a string.
         -> Allows to specify the path to the configuration file
       * options.postcssOptions.config should be a boolean.
         -> Enables/Disables autoloading config
    at validate (/Users/andrew/Projects/Study/nuxt/nuxt-windi/node_modules/schema-utils/dist/validate.js:104:11)
    at Object.loader (/Users/andrew/Projects/Study/nuxt/nuxt-windi/node_modules/postcss-loader/dist/index.js:43:29)

 @ ./node_modules/vue-style-loader??ref--3-oneOf-1-0!./node_modules/css-loader/dist/cjs.js??ref--3-oneOf-1-1!./node_modules/@nuxt/webpack/node_modules/vue-loader/lib/loaders/stylePostLoader.js!./node_modules/postcss-loader/dist/cjs.js??ref--3-oneOf-1-2!./node_modules/@nuxt/webpack/node_modules/vue-loader/lib??vue-loader-options!./.nuxt/components/nuxt-error.vue?vue&type=style&index=0&lang=css& 4:14-378 15:3-20:5 16:22-386
 @ ./.nuxt/components/nuxt-error.vue?vue&type=style&index=0&lang=css&
 @ ./.nuxt/components/nuxt-error.vue
 @ ./.nuxt/index.js
 @ ./.nuxt/client.js
 @ multi ./node_modules/eventsource-polyfill/dist/browserify-eventsource.js (webpack)-hot-middleware/client.js?reload=true&timeout=30000&ansiColors=&overlayStyles=&path=%2F__webpack_hmr%2Fclient&name=client ./.nuxt/client.js
```

**Pull request merged**